x-pack/plugins/observability/public/hooks/slo/use_create_slo.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState } from 'react';
+import type { CreateSLOParams } from '@kbn/slo-schema';
+
+import { useKibana } from '../../utils/kibana_react';
+
+interface UseCreateSlo {
+  loading: boolean;
+  success: boolean;
+  error: string | undefined;
+  createSlo: (slo: CreateSLOParams) => void;
+}
+
+export function useCreateSlo(): UseCreateSlo {
+  const { http } = useKibana().services;
+  const [loading, setLoading] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const createSlo = useCallback(
+    async (slo: CreateSLOParams) => {
+      setLoading(true);
+      setError('');
+      setSuccess(false);
+      const body = JSON.stringify(slo);
+
+      try {
+        await http.post<string>(`/api/observability/slos`, { body });
```

Comment:
Don't we have a type for the create response? If not we should create one. The expected response is not a string but `{id: "uuid"}`